app/models/katello/rhel_lifecycle_status.rb

```diff
@@ -0,0 +1,175 @@
+module Katello
+  class RhelLifecycleStatus < HostStatus::Status
+    UNKNOWN = 0
+    FULL_SUPPORT = 1
+    MAINTENANCE_SUPPORT = 2
+    EXTENDED_SUPPORT = 3
+    APPROACHING_END_OF_SUPPORT = 4
+    SUPPORT_ENDED = 5
+
+    RHEL_EOS_SCHEDULES = { # dates that each support category ends
+      'RHEL9' => {
+        'full_support' => Date.parse('2027-05-31'),
+        'maintenance_support' => Date.parse('2032-05-31'),
+        'extended_support' => Date.parse('2035-05-31')
+      },
+      'RHEL8' => {
+        'full_support' => Date.parse('2024-05-31'),
+        'maintenance_support' => Date.parse('2029-05-31'),
+        'extended_support' => Date.parse('2032-05-31')
+      },
+      'RHEL7' => {
+        'full_support' => Date.parse('2019-08-06'),
+        'maintenance_support' => Date.parse('2024-06-30'),
+        'extended_support' => Date.parse('2028-06-30')
+      },
+      'RHEL7 (System z (Structure A))' => {
+        'full_support' => Date.parse('2019-08-06'),
+        'maintenance_support' => Date.parse('2021-05-31')
+      },
+      'RHEL7 (ARM)' => {
+        'full_support' => Date.parse('2019-08-06'),
+        'maintenance_support' => Date.parse('2020-11-30')
+      },
+      'RHEL7 (POWER9)' => {
+        'full_support' => Date.parse('2019-08-06'),
+        'maintenance_support' => Date.parse('2021-05-31')
+      },
+      'RHEL6' => {
+        'full_support' => Date.parse('2016-05-10'),
+        'maintenance_support' => Date.parse('2020-11-30'),
+        'extended_support' => Date.parse('2024-06-30')
+      },
+      'RHEL5' => {
+        'full_support' => Date.parse('2009-03-31'),
+        'maintenance_support' => Date.parse('2012-11-30'),
+        'extended_support' => Date.parse('2017-11-30')
+      }
+    }.freeze
+
+    EOS_WARNING_THRESHOLD = 1.year
+
+    def self.status_map
+      map = {
+        full_support: FULL_SUPPORT,
+        maintenance_support: MAINTENANCE_SUPPORT,
+        extended_support: EXTENDED_SUPPORT,
+        approaching_end_of_support: APPROACHING_END_OF_SUPPORT,
+        support_ended: SUPPORT_ENDED
+      }
+
+      map.default = UNKNOWN
```

Comment:
I think a RHEL host must be in one of the `full_support`, `maintenance_support`, `extended_support` and `support_ended` states at any given point of time. 
And optionally may be also in `approaching_end_of_support` state when one of the 3 end dates is approaching.
Don't think `UNKNOWN` state is needed here.

Reply:
All of our other host statuses have an `UNKNOWN`. And for non-RHEL hosts, the status won't be one of those three, so it made sense to me to calculate that as `UNKNOWN`.